tensorflow/compiler/mlir/lite/transforms/optimize.cc

```diff
@@ -438,26 +438,26 @@ struct FuseFullyConnectedAndMul : public OpRewritePattern<TFL::MulOp> {
       return failure();
     if (fc_op.fused_activation_function() != "NONE") return failure();
 
-    // Broadcast the constant operand of Mul if it isn't compatible to the
-    // filter input. We only support broadcasting the operand along the depth
-    // dimension, when the operand's depth is 1.
-    Value new_const_val = constant_val;
-    if (!IsBroadcastableElementsAttrAndType(cst.getType(), filter.getType())) {
-      auto original_shape = cst.getType().getShape();
-      llvm::SmallVector<int64_t, 4> normalized_shape(original_shape.begin(),
-                                                     original_shape.end());
-      normalized_shape.push_back(1);
-      auto new_cst = cst.reshape(RankedTensorType::get(
-          normalized_shape, cst.getType().getElementType()));
-      Type new_type = new_cst.getType();
-      if (!IsBroadcastableElementsAttrAndType(new_type, filter.getType())) {
-        return failure();
-      }
-      auto new_op =
-          rewriter.create<ConstantOp>(mul_op.getLoc(), new_type, new_cst);
-      new_const_val = new_op.getResult();
+    // Only fuse multiplier if all dimensions other than the depth dimension
+    // are equal to 1.
```

Comment:
Expand comment on why we only fuse in this case.

Reply:
Thanks for the reply, I tried to improve the comment in 2b2b3a9bd817b05e2451b2de03f170b3282d63f5.

The reason for this is that we can only fuse a multiplier in dimensions that are not summed over during the matmul due to:
![Screenshot 2020-10-20 at 18 07 56](https://user-images.githubusercontent.com/13285808/96620765-c40bcc00-12ff-11eb-9869-37f0b691ed16.png)